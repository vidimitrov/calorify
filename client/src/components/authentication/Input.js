import styled from 'styled-components';

const Input = styled.input`
  background-color: #f6f6f6;
  border: none;
  color: #0d0d0d;
  padding: 15px 32px;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 5px;
  width: 70%;
  border: ${props => (props.invalid ? '2px solid #ff387b' : '2px solid #f6f6f6')} ;
  -webkit-transition: all 0.5s ease-in-out;
  -moz-transition: all 0.5s ease-in-out;
  -ms-transition: all 0.5s ease-in-out;
  -o-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  -webkit-border-radius: 5px 5px 5px 5px;
  border-radius: 5px 5px 5px 5px;
  outline: none;

  &:focus {
    box-shadow: 0 0 5px #ff9a00;
  }
`;

export default Input;
