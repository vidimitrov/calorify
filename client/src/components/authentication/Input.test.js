import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Input from './Input';

it('Authentication/Input component renders correctly', () => {
  const tree = renderer.create(
    <Input value="some-value" placeholder="Add some value here" />,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
