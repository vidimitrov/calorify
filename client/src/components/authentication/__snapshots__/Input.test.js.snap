// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authentication/Input component renders correctly 1`] = `
.c0 {
  background-color: #f6f6f6;
  border: none;
  color: #0d0d0d;
  padding: 15px 32px;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  font-size: 16px;
  margin: 5px;
  width: 70%;
  border: 2px solid #f6f6f6;
  -webkit-transition: all 0.5s ease-in-out;
  -moz-transition: all 0.5s ease-in-out;
  -ms-transition: all 0.5s ease-in-out;
  -o-transition: all 0.5s ease-in-out;
  -webkit-transition: all 0.5s ease-in-out;
  transition: all 0.5s ease-in-out;
  -webkit-border-radius: 5px 5px 5px 5px;
  border-radius: 5px 5px 5px 5px;
  outline: none;
}

.c0:focus {
  box-shadow: 0 0 5px #ff9a00;
}

@media (max-width:420px) {
  .c0 {
    width: 60%;
  }
}

<input
  className="c0"
  placeholder="Add some value here"
  value="some-value"
/>
`;
